### ISSUE TITLE
[ban-types] False positive with custom Function type

### ISSUE THREAD

**Issue by rockwotj** [typescript-eslint/typescript-eslint#3625]
- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/ban-types": ["error"]
  }
}
```

```TS
type Function<A, B> = (a: A) => B;
type Predicate<A> = Function<A, boolean>;
function map<A, B>(a: A[], mapper: Function<A, B>): B[] {
  return a.map(mapper);
}
```

**Expected Result**

Linting to have no errors.

**Actual Result**

There are errors about not using the `Function` type.

```
3:21 - Don't use `Function` as a type. The `Function` type accepts any function-like value. It provides no type safety when calling the function, which can be a common source of bugs. It also accepts things like class declarations, which will throw at runtime as they will not be called with `new`. If you are expecting the function to accept certain arguments, you should explicitly define the function shape.(@typescript-eslint/ban-types)
4:36 - Don't use `Function` as a type. The `Function` type accepts any function-like value. It provides no type safety when calling the function, which can be a common source of bugs. It also accepts things like class declarations, which will throw at runtime as they will not be called with `new`. If you are expecting the function to accept certain arguments, you should explicitly define the function shape.(@typescript-eslint/ban-types)
```

**Additional Info**

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `4.26.0` |
| `@typescript-eslint/parser`        | `4.26.0` |
| `TypeScript`                       | `4.1.2` |
| `ESLint`                           | `7.12.1` |
| `node`                             |  `15.14.0` |

**Comment by bradzacher** [comment]
you're shadowing a global TS type.
You should *not* do this as a general rule. Shadowing global things is a recipe for confusion in your codebase.

---

As for the rule - this is working as expected.
The rule simply looks for a type name and reports on it. That's it - that's the rule.
Doesn't matter if it's global or imported or locally defined. If it's named `Function`, then the default config will report on it.
